src/resolve/sourceComponent.ts

```diff
@@ -129,6 +117,29 @@ export class SourceComponent implements MetadataComponent {
     this.markedForDelete = asDeletion;
   }
 
+  private calculateRelativePath(fsPath: string): string {
+    const { directoryName, suffix, inFolder, folderType } = this.type;
+    // if there isn't a suffix, assume this is a mixed content component that must
+    // reside in the directoryName of its type. trimUntil maintains the folder structure
+    // the file resides in for the new destination.
+    if (!suffix) {
+      return trimUntil(fsPath, directoryName);
+    }
+    // legacy version of folderType
+    if (inFolder) {
+      return join(directoryName, this.fullName.split('/')[0], basename(fsPath));
+    }
+    if (folderType) {
+      // types like Territory2Model have child types inside them.  We have to preserve those folder structures
+      if (this.parentType?.folderType && this.parentType?.folderType !== this.type.id) {
+        const fsPathSplits = fsPath.split(sep);
```

Comment:
I'll echo the same comment from above about using `path.sep` to split file paths.  For this to work we have to normalize the path first unless we are absolutely sure this path can only come from the file system of the OS.